xmppserver/src/main/java/org/jivesoftware/util/cache/DefaultExternalizableUtil.java

```diff
@@ -0,0 +1,471 @@
+/*
+ * Copyright (C) 2020 Ignite Realtime Foundation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.util.cache;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.Externalizable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.ObjectStreamClass;
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Default serialization strategy.
+ *
+ * @author Tom Evans
+ * @author Gaston Dombiak
+ */
+// Note that this implementation is heavily based on the implementation provided by the 'clustering' plugin in
+// org.jivesoftware.openfire.plugin.util.cache.ClusterExternalizableUtil - if changes are made to either implementation,
+// it's likely that the other implementation needs a similar change. See https://issues.igniterealtime.org/browse/OF-1984
```

Comment:
Currently, the HZ implementation swaps out the default one with its own. 

I'd suggest that rather than maintain two identical classes (other than class name) that do the same thing, how about simply making ExternalizableUtil#setStrategy a deprecated, null operation and just use a single class (and ultimately remove the HZ one when support for older Openfire is dropped)

Reply:
I'd be in favor of that, but, at least for now, both classes are not _exact_ duplicates. The Hazelcast one has one specific reference to Hazelcast implementation in it. I'm not sure what breaks if we effectively remove that.

Reply:
Ah, yes. Missed that. Would have to replace `HazelcastInstance.class.getClassLoader()` with introspection (as Openfire won't have access to the HZ classes).

Reply:
That seems like a bridge to far for now. Maybe we can make the implementation in the plugin extend this one to reduce code maintenance, after we bump the Openfire version dependency at some point?